### PR TITLE
GEODE-5172: refactor txApplyPut to reuse RegionMapPut

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
AbstractRegionMapPut has been introduced and has the common code used for both a non-tx put (RegionMapPut) and a tx put (RegionMapCommitPut).
RegionMapCommitPut is used by txApplyPut. I thought this name was better than RegionMapTransactionPut because this is not a put done during a transaction but what needs to be done at commit time for a put that was done during a transaction. Another possible name would be RegionMapTransactionApplyPut but I liked CommitPut better because it is more concise.

The unit tests provide good coverage for both AbstractRegionMapPut and RegionMapCommitPut.
The old unit test for RegionMapPut still exists and has been improved but still does not cover all the code in RegionMapPut class. I will file a ticket to improve that test. At one time I thought that would be part of this ticket but I think this ticket should focus on txApplyPut.

@dhemery

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/map/AbstractRegionMapPut.java [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+  private boolean clearOccurred;
+  private long lastModifiedTime;
+  private RegionEntry regionEntry;
+  private boolean create;
+  private boolean completed;
+
+  public AbstractRegionMapPut(FocusedRegionMap focusedRegionMap, InternalRegion owner,
+      EntryEventImpl event) {
+    if (owner == null) {
+      Assert.assertTrue(false, "The owner for RegionMap " + this + " is null for event " + event);
```
Comment (dhemery) [inline geode-core/src/main/java/org/apache/geode/internal/cache/map/AbstractRegionMapPut.java]: `assertTrue(false, …)` is weird. Consider: `Assert.fail(…)`. Or better yet, replace the entire `if` with:

    Objects.requireNonNull(owner, () -> "The owner for RegionMap " + this + " is null for event " + event);

This style of assertion has the nice side effect of making IntelliJ relax about the later use of `owner`.

Comment (dhemery) [reply geode-core/src/main/java/org/apache/geode/internal/cache/map/AbstractRegionMapPut.java]: Add a unit test for a null owner.

File: geode-core/src/main/java/org/apache/geode/internal/cache/map/AbstractRegionMapPut.java [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    }
+    this.focusedRegionMap = focusedRegionMap;
+    this.owner = owner;
+    this.event = event;
+    this.ownerInitialized = owner.isInitialized();
+  }
+
+  protected boolean isOwnerUninitialized() {
+    return !ownerInitialized;
```
Comment (dhemery) [inline geode-core/src/main/java/org/apache/geode/internal/cache/map/AbstractRegionMapPut.java]: Consider eliminating this method. One way to do it is to implement it as:

    return !isOwnerInitialized();

Then inline all uses of `isOwnerUninitialized()`
